### DISCUSSION TITLE
Is it possible to use S3 for config instead of Minio?

### DISCUSSION THREAD

**Discussion by dougwettlaufer** [Budibase/budibase#5430] [Help] [answered]
In the helm chart, under `objectStore` I see references to `region` and `AWS_ACCESS_KEY` but I'm a little lost on setting the bucket. From the looks of things it seems like the bucket names are hardcoded in utils.js

```js
exports.ObjectStoreBuckets = {
  BACKUPS: "backups",
  APPS: "prod-budi-app-assets",
  TEMPLATES: "templates",
  GLOBAL: "global",
  GLOBAL_CLOUD: "prod-budi-tenant-uploads",
}
```

Since a bucket in S3 must be globally unique this won't work. Please excuse my lack of JS knowledge but from grepping through the code it doesn't seem like there's a way to override the bucket name or add a unique prefix. Can someone please confirm that's the case?

**Comment by shogunpurple** [comment, marked as answer]
Great catch @dougwettlaufer - you are right. It works fine for MinIO and fine for us in our production cloud because we own that bucket, but will not work for self-hosted kubernetes users who are using S3. I'll convert this to an issue and we will get it addressed very soon.

Thanks!

**Reply by dougwettlaufer** [reply]
Thanks @shogunpurple !
